t/01-test_needle.t

```diff
@@ -337,6 +337,33 @@ is($res->{needle}->{name}, 'login_sddm.ref.perfect', "perfect match should win")
 $res = $img1->search([$imperfect, $workaround], 0.9, 0);
 is($res->{needle}->{name}, 'login_sddm.ref.workaround.imperfect', "workaround match should win");
 
+# test caching via needle->get_image
```

Comment:
because you add this header for this section, why not move it into a subtest?

Reply:
ok, will you?

Reply:
Not as part of this PR. The benefit wouldn't be that high in my opinion, e.g. the different sub tests would actually still be intertwined.